magicschoolbus/Assets/_Code/FreelookRubberBand.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Cinemachine;
+using UnityEngine;
+
+// Written by Dreaming I'm Latios
+// If this is not sufficient, please let me know. It can be improved.
+// Also, the code is not fully optimized. It shouldn't be a big deal since there should only be a small number of these running.
+// Usage: Leave the freelook's x-axis value range between -180 and 180. Also do not rotate the freelook's gameobject.
+
+// Fascination = What the camera is tracking.
+
+namespace MSB
+{
+    [RequireComponent(typeof(CinemachineFreeLook))]
+    public class FreelookRubberBand : MonoBehaviour
+    {
+        [Tooltip("Higher values will cause the camera to auto-orbit more aggressively.")]
+        public float                m_followBias         = 1f;
+        public float                m_autoOrbitMaxSpeed  = 45f;
+        public float                m_inputOrbitSpeed    = 60f;
+        public string               m_bindingXAxis       = "";
+        public string               m_bindingResetButton = "";
+        private CinemachineFreeLook m_Freelook;
+        private Vector3             m_prevTargetPosition;
+
+        private void Awake()
+        {
+            m_Freelook                          = GetComponent<CinemachineFreeLook>();
+            m_prevTargetPosition                = m_Freelook.m_Follow.position;
+            m_Freelook.m_BindingMode            = CinemachineTransposer.BindingMode.WorldSpace;
+            m_Freelook.m_XAxis.m_InputAxisName  = "";
+            m_Freelook.m_XAxis.m_InputAxisValue = 0f;
+        }
+
+        private void LateUpdate()
+        {
+            if (m_bindingXAxis.Length > 0)
+                m_Freelook.m_XAxis.Value += Input.GetAxis(m_bindingXAxis) * m_inputOrbitSpeed * Time.deltaTime;
+            if (m_bindingResetButton.Length > 0)
+            {
+                if (Input.GetButtonDown(m_bindingResetButton))
+                {
+                    m_Freelook.m_YAxis.Value = 0.5f;
+                    Vector3 targetForward    = m_Freelook.m_Follow.forward;
+                    targetForward.y          = 0f;
+                    m_Freelook.m_XAxis.Value = Vector3.SignedAngle(transform.forward, targetForward, Vector3.up);
+                }
+            }
+
+            // Unity API uses positive angles in clockwise direction. This makes the trig confusing.
+            // So I reverse the sign immediately and then reverse it back on writeback.
+            Vector3 currTargetPosition = m_Freelook.m_Follow.position;
+            Vector3 deltaPosition      = currTargetPosition - m_prevTargetPosition;
+            deltaPosition.y            = 0f;
+            // If our fascination isn't moving, neither should the camera.
```

Comment:
We might want to keep adjusting the camera back to behind our character at a minimum lerp speed unless the camera is unlocked in a true freelook mode orbiting around the player using mouse input.  Or maybe make this a toggle?  I don't think we necessarily want to stop rotating the camera back when the character stops.

Reply:
This is the correct behavior for gamepad controls, which is what I was testing. I don't play 3rd person adventure games on mouse and keyboard so I have no idea what proper behavior is for those games. However, I will add an override so you can drive the input values via script instead of input so that you can handle the additional logic required for mouse and keyboard. 

Also, "we might want" is just unnecessary scope creep. Either you want it or you don't. And if you don't know, then you need to figure that out. I'm not at your meetings. I can't tell you what you want. And there's no such thing as future-proof. Don't try to stall an oncoming train because you are uncertain which side of the tracks you want to be on.